### PR TITLE
computing norms

### PR DESCRIPTION
**Pull request by denisergashbaev**

### CODE REVIEW

File: ptb-lm.py [hunk 1]
```diff
@@ -330,6 +334,10 @@ def subsequent_mask(size):
 else:
     print("Model type not recognized.")
 
+#todo new code to load model
```
Comment (denisergashbaev) [inline ptb-lm.py:337]: @ggegiya1 this is how we load the model

File: models.py [hunk 2]
```diff
@@ -63,7 +62,9 @@ def forward(self, inputs, hidden):
         """
         hidden = self.h2h(hidden) + self.i2h(inputs)
         hidden = self.non_linearity(hidden)
-
+        # no gradient is computed if we don't call requires_grad_(True)
```
Comment (denisergashbaev) [inline models.py:65]: for 5.2 we need to compute the gradient with respect to the hidden state. So i activate the gradient here

File: models.py [hunk 3]
```diff
@@ -212,29 +213,27 @@ def forward(self, inputs, hidden):
...
                 # apply dropout to the vertical outputs
                 layer_output = self.dropout(layer_hidden)  # shape (batch_size, hidden_size)
-
                 # save output
                 hidden_list.append(layer_hidden)
 
-            # update hidden state after processing all layers for a single batch (num_layers, seq_len, hidden_size)
-            hidden = torch.stack(hidden_list)
+            # update hidden state after processing all layers for a single batch
+            # (num_layers, seq_len, hidden_size)
```
Comment (denisergashbaev) [inline models.py:228]: and save the hiddens in the list to iterate over them after the backward pass

File: ptb-lm.py [hunk 4]
```diff
@@ -413,7 +421,21 @@ def run_epoch(model, data, is_train=False, lr=1.0):
         iters += model.seq_len
         if args.debug:
             print(step, loss)
-        if is_train:  # Only update parameters if training 
+
+        # todo computing gradient norms
+        if args.compute_gradient:
```
Comment (denisergashbaev) [inline ptb-lm.py:428]: and on the validation we iterate over the hidden states and get their gradients, turn them to norms
